### PR TITLE
user show edit delete question

### PR DESCRIPTION
**Pull request by trikmbk56**

### CODE REVIEW

**Comment by NeverSmileK57CLC** [comment]
OK

File: app/controllers/questions_controller.rb [hunk 1]
```diff
@@ -11,17 +13,45 @@ def new
 
   def create
     if @question.save
-      flash[:sucess] = t "user.question.create_success"
+      flash[:success] = t "user.question.create_success"
+      redirect_to :back
     else
       flash[:danger] = t "user.question.create_fail"
+      render :create
```
Comment (hellovietnam93) [inline app/controllers/questions_controller.rb]: ???

File: app/views/questions/_question.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<tr>
+  <td><%= question.id %></td>
+  <td><%= link_to question.content, user_question_path(user, question) %></td>
+  <td><%= question.subject.name %></td>
+  <td><%= question.question_type %></td>
+  <td>
+    <% if question.waiting? %>
+      <span class="label label-info"><%= question.state %></span>
+    <% elsif question.aeecpt? %>
```
Comment (hellovietnam93) [inline app/views/questions/_question.html.erb]: aeecpt ?

File: app/views/questions/index.html.erb [hunk 3]
```diff
@@ -6,5 +5,23 @@
...
+          <th><%= t "user.question.question" %></th> 
+          <th><%= t "user.question.subject" %></th> 
+          <th><%= t "user.question.type" %></th> 
+          <th><%= t "user.question.state" %></th> 
+          <th></th>
+        </tr>
+      </thead>
+      <tbody>
+        <ul class="questions">
+          <%= render @questions, user: @user %>
```
Comment (hellovietnam93) [inline app/views/questions/index.html.erb]: need ```, user: @user``` ?

File: app/views/questions/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        <tr>
+          <th scope="row"><%= t "user.question.answer" %></th>
+          <td>
+            <table>
+              <tbody>
+                <% @question.answers.each do |answer| %>
+                <tr>
+                  <td>
+                    <% if answer.correct %>
+                    <span class="label label-success"><%= answer.correct %></span>
```
Comment (hellovietnam93) [inline app/views/questions/show.html.erb]: ident

File: app/views/questions/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+        <tr>
+          <th scope="row"><%= t "user.question.type" %></th>
+          <td><%= @question.question_type %></td>
+        </tr>
+
+        <tr>
+          <th scope="row"><%= t "user.question.state" %></th>
+          <td>
+            <% if @question.waiting? %>
```
Comment (hellovietnam93) [inline app/views/questions/show.html.erb]: write a helper method

File: app/views/questions/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+          </td>
+        </tr>
+      </tbody>
+    </table>
+
+    <div class="row">
+      <%= link_to t("user.question.edit"), edit_user_question_path(current_user, @question),
+        class: "btn btn-info glyphicon glyphicon-pencil" %>
+      <%= link_to t("user.question.delete"), [current_user, @question], method: :delete,
+          data: {confirm: t("you_sure")}, class: "btn btn-danger  glyphicon glyphicon-trash" %>
```
Comment (hellovietnam93) [inline app/views/questions/show.html.erb]: ident

File: app/controllers/questions_controller.rb [hunk 7]
```diff
@@ -11,17 +13,45 @@ def new
...
+    else
+      flash[:danger] = t "user.question.update_fail"
+      render :edit
+    end
+  end
+
+  def destroy
+    @question.destroy
+    flash[:success] = t "user.question.delete_success"
+    redirect_to user_questions_path
```
Comment (hellovietnam93) [inline app/controllers/questions_controller.rb]: no k bi loi ah, thieu user ma nhi

Comment (trikmbk56) [inline app/controllers/questions_controller.rb]: nếu thêm user vào thì nó sẽ bị lỗi

File: app/views/questions/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+        <tr>
+          <th scope="row"><%= t "user.question.answer" %></th>
+          <td>
+            <table>
+              <tbody>
+                <% @question.answers.each do |answer| %>
+                <tr>
+                  <td>
+                    <% if answer.correct %>
```
Comment (hellovietnam93) [inline app/views/questions/show.html.erb]: refactor

File: app/views/questions/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+          <td>	<%= question_state @question %></td>
+        </tr>
+      </tbody>
+    </table>
+
+    <div class="row">
+      <%= link_to t("user.question.edit"), edit_user_question_path(current_user, @question),
+        class: "btn btn-info glyphicon glyphicon-pencil" %>
+      <%= link_to t("user.question.delete"), [current_user, @question], method: :delete,
+          data: {confirm: t("you_sure")}, class: "btn btn-danger glyphicon glyphicon-trash" %>
```
Comment (hellovietnam93) [inline app/views/questions/show.html.erb]: ident

File: app/helpers/questions_helper.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+module QuestionsHelper
+  def question_state question
+    if question.waiting?
```
Comment (hellovietnam93) [inline app/helpers/questions_helper.rb]: case when

**Comment by hellovietnam93** [comment]
OK

File: app/controllers/questions_controller.rb [hunk 11]
```diff
@@ -11,17 +13,45 @@ def new
...
   private
   def question_params
     params.require(:question).permit :question_type, :content, :subject_id,
-      :created_by, :state, answers_attributes: [:id,
+      :user_id, :state, answers_attributes: [:id,
       :correct, :content, :_destroy]
   end
+
+  def load_user
+    @user = User.find params[:user_id]
```
Comment (dieunb) [inline app/controllers/questions_controller.rb]: Sao phải load user hả em?

**Pull request merged**